deepchem/models/tensorgraph/layers.py

```diff
@@ -91,6 +91,8 @@ def _get_input_tensors(self, in_layers, reshape=False):
     for input in in_layers:
       if isinstance(input, tf.Tensor):
         tensors.append(input)
+      elif isinstance(input, tf.Variable):
+        tensors.append(input)
```

Comment:
And *why* is it that variables aren't tensors?  I don't know.  They probably had some reason for designing it like that, but really...

Knowing a little bit more about Tensorflow than I did when I first wrote this code, I think we could probably replace this series of ifs with `tensors.append(tf.convert_to_tensor(input))`.  That will work for all these cases (layers, tensors, variables), but also support anything else that can be converted to a tensor (number, numpy array, etc.)